packages/lit-dev-content/site/docs/v3/releases/upgrade.md

```diff
@@ -1,40 +1,63 @@
 ---
-title: Upgrade guide
+title: Lit 3 upgrade guide
 eleventyNavigation:
-  key: Upgrade guide
+  key: Lit 3 upgrade guide
   parent: Releases
   order: 3
 versionLinks:
   v2: releases/upgrade/
 ---
 
-## Overview
+<div class="alert alert-info">
+
+If you are looking to migrate from Lit 1.x to Lit 2.x, see the [Lit 2 upgrade guide](/docs/v2/releases/upgrade/).
 
-Lit 3.0 has very few breaking changes from Lit 2.0. Any projects using Lit 2
-without deprecation warnings should not require any changes to upgrade to Lit 3.
+</div>
 
-Lit 3.0:
+## Overview
 
-- Drops support for IE11
-- Is published as ES2021
-- Removes a few deprecated APIs
+Lit 3.0 has very few breaking changes from Lit 2.x:
 
-Lit 3.0 should require no changes to upgrade from Lit 2.0 for the vast majority
+- IE11 is no longer supported.
+- Lit's npm modules are now published as ES2021.
+- APIs marked deprecated in Lit 2.x releases have been removed.
+- SSR hydration support modules have moved to the `@lit-labs/ssr-client` package.
+- Decorator behavior has been unified between TypeScript experimental decorators and standard decorators.
+- Support was removed for Babel decorators version "2018-09"
+
+Lit 3.0 should require no changes to upgrade from Lit 2.x for the vast majority
 of users. Most apps and libraries should be able to extend their npm version
 ranges to include both 2.x and 3.x, like `"^2.7.0 || ^3.0.0"`.
 
-Lit 2.x and 3.0 are _interoperable_: templates, base classes, directives,
-decorators, etc., from one version of Lit will work with those from another.
+Lit 2.x and 3.0 are _interoperable_: templates, base classes, and directives from one version of Lit will work with those from another.
+
+## Decorator behavior has been unified
+
+In Lit 3.0 our decorators support running in three configurations:
+
+1. With the TypeScript `experimentalDecorators` flag and properties without the `accessor` keyword. This is backwards compatible with Lit 2 users. As standard decorators are still immature, this is the most performant option that we recommend for production for the time being.
+2. With the TypeScript `experimentalDecorators` flag and with the `accessor` keyword. This aids migration to standard decorators. Code can add the `accessor` keyword one decorator at a time to migrate gradually.
+3. As standard decorators, which need the `accessor` keyword, and TypeScript 5.2 (without the `experimentalDecorators` flag), or Babel 7.23.0.
```

Comment:
I think it's potentially a bit confusing to call this three configurations.

Configuration-wise, it's really 2 cases:
1. TypeScript experimental decorators (with `"experimentalDecorators": true` and `"useDefineForClassFields": false` in tsconfig)
2. TC39 standard decorators with metadata (available with TypeScript 5.2 or Babel 7.23.0)

(Then we should skip to "To make Lit decorators consistent..." with the list of breaking changes.)

To use the standard decorator configuration, `@property()` decorated class fields **must** include the `accessor` keyword. (Link to a new entry in "## Steps to Upgrade" section showing code change needed.)

To aid in migration from experimental decorators to standard decorators, decorated class fields with the `accessor` keyword will also work with the `experimentalDecorators` flag set to true. So users can start migrating code without changing the TS configuration.

(Then we should include the [snippet suggested by Peter](https://github.com/lit/lit.dev/pull/1199/files#r1344552867) about recommending users stick to experimental decorators for the time being.)

Reply:
Yeah, configuraitons is a bit confusing, it suggests a build tool config. I think we should document these three cases though. What about:

```markdown
In Lit 3.0 there are three supported ways to use our decorators:
```

Reply:
Also, maybe we should put the big idea up front in bold:

```markdown
The Lit 3 decorators are largely backwards compatible. If you've been using them with TypeScript, **most likely no changes are needed**. We've added forward compatibility with standard decorators, but the legacy `experimentalDecorators: true` without the `accessor` keyword is currently the more performant option.
```

Reply:
Hmm.. While I believe we should definitely mention that `accessor` fields work with experimental decorator config, I don't know if I would call it out in a list of 3 "supported ways" like that. In that sense, writing separate getter setter and decorating the setter is also a "supported way" of using our decorators.

Reply:
However, a valid 3rd *configuration* is
```json
{
  "experimentalDecorators": true,
  "useDefineForClassFields": true
}
```
with decorated fields using `accessor`

Reply:
And also speaking of configuration, I think we should make clear that standard decorator support for TypeScript is currently strictly `tsc` only. `esbuild` and `swc` do not yet have all the necessary implementations. This means Vite projects can't have tsconfigs that don't set experimental decorators. Maybe this belongs in the launch blog post, rather than the upgrade guide.. but is certainly another reason to recommend sticking to experimental decorators.